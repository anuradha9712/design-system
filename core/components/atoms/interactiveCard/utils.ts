export const removeItem = (targetItem: any, arr = []) => {
  return arr.filter((item: any) => item !== targetItem);
};
