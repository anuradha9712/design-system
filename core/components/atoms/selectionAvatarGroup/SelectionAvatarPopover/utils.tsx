import React from 'react';

export const handleKeyDown = (
  event: React.KeyboardEvent,
  focusedOption: Element | undefined,
  setFocusedOption?: React.Dispatch<React.SetStateAction<Element | undefined>>,
  setHighlightFirstItem?: React.Dispatch<React.SetStateAction<boolean>>,
  setHighlightLastItem?: React.Dispatch<React.SetStateAction<boolean>>,
  listRef?: any,
  withSearch?: boolean
) => {
  switch (event.key) {
    case 'ArrowUp':
      event.preventDefault();
      navigateOptions('up', focusedOption, setFocusedOption, listRef, withSearch);
      break;
    case 'ArrowDown':
      event.preventDefault();
      navigateOptions('down', focusedOption, setFocusedOption, listRef, withSearch);
      break;
    case 'Enter':
      handleEnterKey(focusedOption);
      setHighlightLastItem?.(false);
      setHighlightFirstItem?.(false);
      break;
    default:
      break;
  }
};

const handleEnterKey = (focusedOption: Element | undefined) => {
  (focusedOption as HTMLElement)?.click();
};

const navigateSelectedOption = (
  index: number,
  withSearch?: boolean,
  listItems?: any,
  direction?: string,
  listRef?: any,
  setFocusedOption?: React.Dispatch<React.SetStateAction<Element | undefined>>
) => {
  console.log('tttindexxx', index);
  // if (withSearch) {
  //   const searchInput = listRef.current.querySelector('[data-test="DesignSystem-SelectionAvatar--Input"]');
  //   console.log('ttt inside iff');
  //   if ((index === 0 && direction === 'up') || (index === listItems.length - 2 && direction === 'down')) {
  //     console.log('tttfocus search');
  //     searchInput.focus();
  //     setFocusedOption && setFocusedOption(searchInput);
  //     return;
  //   }
  // }
  const searchInput = listRef.current.querySelector('[data-test="DesignSystem-SelectionAvatar--Input"]');
  searchInput.focus();
  setFocusedOption && setFocusedOption(searchInput);
};

const navigateOptions = (
  direction: string,
  focusedOption: Element | undefined,
  setFocusedOption?: React.Dispatch<React.SetStateAction<Element | undefined>>,
  listRef?: any,
  withSearch?: boolean
) => {
  const listItems = listRef.current.querySelectorAll('[data-test="DesignSystem-Listbox-ItemWrapper"]');
  let index = Array.from(listItems).findIndex((item) => {
    return item == focusedOption;
  });

  console.log('tttwithSearchwithSearch', withSearch, 'index', index, 'listItems[index]', listItems[index]);

  if (index === -1) {
    console.log('ttt11');
    index = direction === 'up' ? listItems.length - 1 : 0;
  } else if (
    (withSearch && index === 0 && direction === 'up') ||
    (withSearch && index === listItems.length - 1 && direction === 'down')
  ) {
    console.log('ttt22');

    navigateSelectedOption(index, withSearch, listItems, direction, listRef, setFocusedOption);
    // return;
  } else {
    console.log('ttt333');

    index = direction === 'up' ? (index - 1 + listItems.length) % listItems.length : (index + 1) % listItems.length;

    const targetOption = listItems[index];

    (targetOption as HTMLElement).focus();
    setFocusedOption && setFocusedOption(targetOption);
    targetOption.scrollIntoView({ block: 'center' });
  }
};

export const handleInputKeyDown = (
  event: React.KeyboardEvent,
  listRef: any,
  setFocusedOption: React.Dispatch<React.SetStateAction<Element | undefined>>
) => {
  const listItems = listRef.current?.querySelectorAll('[data-test="DesignSystem-Listbox-ItemWrapper"]');
  let targetOption;

  switch (event.key) {
    case 'ArrowUp':
      event.preventDefault();
      targetOption = listItems[listItems.length - 1];
      break;
    case 'ArrowDown':
      event.preventDefault();
      targetOption = listItems[0];
      break;
    default:
      break;
  }

  (targetOption as HTMLElement)?.focus();
  targetOption?.scrollIntoView({ block: 'center' });
  setFocusedOption && setFocusedOption(targetOption);
};
