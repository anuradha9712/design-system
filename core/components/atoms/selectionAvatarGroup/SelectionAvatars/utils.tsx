import React from 'react';

export const handleKeyDown = (
  event: React.KeyboardEvent,
  setOpenPopover?: React.Dispatch<React.SetStateAction<boolean>>,
  setHighlightFirstItem?: React.Dispatch<React.SetStateAction<boolean>>,
  setHighlightLastItem?: React.Dispatch<React.SetStateAction<boolean>>
) => {
  switch (event.key) {
    case 'Enter':
      setOpenPopover?.(true);
      setHighlightFirstItem?.(true);
      break;
    case 'ArrowDown':
      setOpenPopover?.(true);
      setHighlightFirstItem?.(true);
      break;
    case 'ArrowUp':
      setOpenPopover?.(true);
      setHighlightLastItem?.(true);
      break;
    default:
      break;
  }
};

export const focusListItem = (
  position: string,
  setFocusedOption?: React.Dispatch<React.SetStateAction<Element | undefined>>,
  listRef?: any,
  withSearch?: boolean
) => {
  if (withSearch && position === 'down') {
    const searchInput = listRef.current?.querySelectorAll('[data-test="DesignSystem-SelectionAvatar--Input"]');
    console.log('aa22', searchInput);
    searchInput[0]?.focus();
    setFocusedOption && setFocusedOption(searchInput);
    return;
  }

  const listItems = listRef.current?.querySelectorAll('[data-test="DesignSystem-Listbox-ItemWrapper"]');
  console.log('listReflistRef', listRef.current, 'listItems', listItems);
  let targetOption;

  if (position === 'down') {
    targetOption = listItems[0];
  } else {
    targetOption = listItems[listItems.length - 1];
  }
  (targetOption as HTMLElement)?.focus();
  targetOption?.scrollIntoView({ block: 'center' });
  setFocusedOption && setFocusedOption(targetOption);
};
