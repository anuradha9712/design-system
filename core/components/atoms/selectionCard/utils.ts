export const removeItem = (targetItem: string, arr: string[]) => {
  return arr.filter((item: string) => item !== targetItem);
};

export const handleMultiSelect = (
  selectedOptions: string[],
  isItemSelected: boolean,
  setSelectedOptions: React.Dispatch<React.SetStateAction<string[]>>,
  id: string
) => {
  let resultList = [...selectedOptions];
  if (isItemSelected) {
    resultList = removeItem(id, selectedOptions);
  } else {
    resultList.push(id);
  }
  setSelectedOptions(resultList);
};

export const handleSingleSelect = (
  selectedOptions: string[],
  isItemSelected: boolean,
  setSelectedOptions: React.Dispatch<React.SetStateAction<string[]>>,
  id: string
) => {
  let resultList = [...selectedOptions];
  if (isItemSelected) {
    resultList = [];
  } else {
    resultList = [id];
  }

  setSelectedOptions(resultList);
};
