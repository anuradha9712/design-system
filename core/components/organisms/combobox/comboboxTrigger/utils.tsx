import React from 'react';

export const handleKeyDown = (
  event: React.KeyboardEvent,
  setOpenPopover?: React.Dispatch<React.SetStateAction<boolean>>,
  // setFocusedOption?: React.Dispatch<React.SetStateAction<Element | undefined>>,
  setHighlightFirstItem?: any,
  setHighlightLastItem?: any
) => {
  console.log('handle key down e.key', event.key);

  switch (event.key) {
    case 'ArrowUp':
      setOpenPopover?.(true);
      setHighlightLastItem(true);
      // focusListItem('up', setFocusedOption);
      // requestAnimationFrame(() => focusListItem('up', setFocusedOption));
      break;
    case 'ArrowDown':
      setOpenPopover?.(true);
      setHighlightFirstItem(true);

      // requestAnimationFrame(() => focusListItem('down', setFocusedOption));
      break;
    case 'Escape':
      setOpenPopover?.(false);
      break;
    default:
      break;
  }
};

export const focusListItem = (
  position: string,
  setFocusedOption?: React.Dispatch<React.SetStateAction<Element | undefined>>
) => {
  const listItems = document.querySelectorAll('[data-test="DesignSystem-Listbox-ItemWrapper"]');
  let targetOption;
  // if (position === 'down') {
  //   targetOption = listItems[0];
  //   (listItems[0] as HTMLElement)?.focus();
  // } else {
  //   (listItems[listItems.length - 1] as HTMLElement)?.focus();
  // }

  if (position === 'down') {
    console.log('rrr down', listItems);
    targetOption = listItems[0];
  } else {
    targetOption = listItems[listItems.length - 1];
  }
  console.log('rrrtargetOption', targetOption);
  (targetOption as HTMLElement)?.focus();
  targetOption?.scrollIntoView({ block: 'center' });
  setFocusedOption && setFocusedOption(targetOption);
};
