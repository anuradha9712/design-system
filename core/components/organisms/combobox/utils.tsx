import React from 'react';

export const handleKeyDown = (
  event: React.KeyboardEvent,
  focusedOption: Element | undefined,
  setFocusedOption?: React.Dispatch<React.SetStateAction<Element | undefined>>,
  setOpenPopover?: React.Dispatch<React.SetStateAction<boolean>>,
  inputTriggerRef?: any
) => {
  switch (event.key) {
    case 'ArrowUp':
      event.preventDefault();
      navigateOptions('up', focusedOption, setFocusedOption);
      break;
    case 'ArrowDown':
      event.preventDefault();
      navigateOptions('down', focusedOption, setFocusedOption);
      break;
    case 'Enter':
      handleEnterKey(focusedOption);
      break;
    case 'Escape':
      setOpenPopover?.(false);
      console.log('inputTriggerRef.current', inputTriggerRef.current);
      inputTriggerRef.current.focus();
      break;
    default:
      break;
  }
};

const handleEnterKey = (focusedOption: Element | undefined) => {
  (focusedOption as HTMLElement)?.click();
};

const navigateOptions = (
  direction: string,
  focusedOption: Element | undefined,
  setFocusedOption?: React.Dispatch<React.SetStateAction<Element | undefined>>
) => {
  const listItems = document.querySelectorAll('[data-test="DesignSystem-Listbox-ItemWrapper"]');
  let index = Array.from(listItems).findIndex((item) => {
    return item == focusedOption;
  });

  if (index === -1) {
    index = direction === 'up' ? listItems.length - 1 : 0;
  } else {
    index = direction === 'up' ? (index - 1 + listItems.length) % listItems.length : (index + 1) % listItems.length;
  }

  const targetOption = listItems[index];
  console.log('targetOptiontargetOption', targetOption);
  (targetOption as HTMLElement).focus();
  setFocusedOption && setFocusedOption(targetOption);
  targetOption.scrollIntoView({ block: 'center' });
};
