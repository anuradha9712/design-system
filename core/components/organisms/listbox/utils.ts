const focusOption = (targetIndex: number, totalLength: number, direction: string) => {
  let startIndex = targetIndex;
  const endIndex = direction === 'down' ? totalLength : -1;

  while (startIndex !== endIndex) {
    const selectedElement = document.querySelector(`[data-list-item-index="${startIndex}"]`) as HTMLElement;

    if (selectedElement && selectedElement.getAttribute('data-disabled') !== 'true') {
      selectedElement && selectedElement.focus();
      break;
    }

    if (direction === 'down') {
      startIndex++;
    } else {
      startIndex--;
    }
  }
};

export const onKeyDown = (event: React.KeyboardEvent, totalLength: number) => {
  const sourceElement = event.target as HTMLElement;
  const sourceIndex = sourceElement.getAttribute('data-list-item-index');
  const targetIndex = (sourceIndex && parseInt(sourceIndex, 10)) || 0;

  switch (event.key) {
    case 'ArrowDown':
      event.preventDefault();
      focusOption(targetIndex + 1, totalLength, 'down');
      break;
    case 'ArrowUp':
      event.preventDefault();
      focusOption(targetIndex - 1, totalLength, 'up');
      break;
    case 'Enter': {
      event.preventDefault();
      sourceElement.click();
      break;
    }
    default:
      break;
  }
};
