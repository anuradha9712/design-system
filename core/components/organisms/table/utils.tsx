import { RowData, GridProps } from '../grid';

const isElementPresent = (list: [], uniqueColumnName: string, value: any) => {
  const arr = list?.filter((item) => item[uniqueColumnName] === value);
  return arr.length > 0;
};

export const getUpdatedData = (
  data: GridProps['data'],
  uniqueColumnName: string,
  selectedList: [],
  isSelectAll?: boolean,
  isCancelSelection?: boolean
) => {
  const updatedData = data.map((item: RowData) => {
    if (
      isSelectAll ||
      item._selected ||
      (item[uniqueColumnName] &&
        selectedList &&
        isElementPresent(selectedList, uniqueColumnName, item[uniqueColumnName]) &&
        !isCancelSelection)
    ) {
      item._selected = true;
    } else if (isCancelSelection) {
      item._selected = false;
    } else {
      item._selected = false;
    }

    return item;
  });

  return updatedData;
};

const uniqueByKey = (arr: any[], key: string) => {
  const seen = new Set();
  return arr.filter((obj) => {
    const value = obj[key];
    if (seen.has(value)) {
      return false;
    } else {
      seen.add(value);
      return true;
    }
  });
};

export const removeDuplicate = (arrayOfObjects: any[], uniqueColumnName: string) => {
  console.log('arrayOfObjects arrayOfObjects', arrayOfObjects);
  return uniqueByKey(arrayOfObjects, uniqueColumnName);
};
