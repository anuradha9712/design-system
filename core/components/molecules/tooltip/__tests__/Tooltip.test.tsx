import * as React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { Tooltip, Button } from '@/index';
import { TooltipProps as Props } from '@/index.type';
import { testHelper, filterUndefined, valueHelper, testMessageHelper } from '@/utils/testHelper';

const Position = ['top', 'top-start', 'top-end', 'bottom', 'bottom-start', 'bottom-end', 'left', 'right'];
const StringValue = 'Sample string';
const Size = ['small', 'regular'];

const mapper = {
  position: valueHelper(Position, { required: true, iterate: true }),
  size: valueHelper(Size, { required: true, iterate: true }),
  tooltip: valueHelper(StringValue, { required: true }),
};

describe('Tooltip component', () => {
  const testFunc = (props: Record<string, any>): void => {
    const attr = filterUndefined(props) as Props;
    const { children, ...rest } = attr;

    it(testMessageHelper(attr), () => {
      const { baseElement } = render(
        <Tooltip {...rest}>
          <Button>Button</Button>
        </Tooltip>
      );
      expect(baseElement).toMatchSnapshot();
    });
  };

  testHelper(mapper, testFunc);

  it('should not render tooltip when showTooltip is false', () => {
    const { baseElement, getByRole, queryByText } = render(
      <Tooltip showTooltip={false} tooltip="A tooltip">
        <Button>Button</Button>
      </Tooltip>
    );
    const button = getByRole('button');
    fireEvent.mouseOver(button);
    const tooltipText = queryByText('A tooltip');
    expect(tooltipText).not.toBeInTheDocument();
    expect(baseElement).toMatchSnapshot();
  });
});

describe('Tooltip component with text overflow', () => {
  it('should render tooltip when showOnTruncation is false', () => {
    const { getByRole, queryByText } = render(
      <Tooltip showOnTruncation={false} tooltip="A tooltip">
        <Button style={{ maxWidth: 150 }} className="ellipsis--noWrap d-inline-block w-100">
          show me the tooltip on hover
        </Button>
      </Tooltip>
    );
    const button = getByRole('button');
    fireEvent.mouseOver(button);
    const tooltipText = queryByText('A tooltip');
    expect(tooltipText).toBeInTheDocument();
  });

  it('should render tooltip when showOnTruncation is true with text overflow', () => {
    const { getByRole, queryByText } = render(
      <Tooltip showOnTruncation={true} tooltip="show me the tooltip on hover">
        <Button style={{ maxWidth: 150 }} className="ellipsis--noWrap d-inline-block w-100">
          show me the tooltip on hover
        </Button>
      </Tooltip>
    );
    const button = getByRole('button');
    fireEvent.mouseOver(button);
    const tooltipText = queryByText('show me the tooltip on hover');
    expect(tooltipText).toBeInTheDocument();
  });

  it('should not render tooltip when showOnTruncation is true and no text overflow', () => {
    const { getByRole, queryByText } = render(
      <Tooltip showOnTruncation={true} tooltip="A tooltip">
        <Button style={{ maxWidth: 300 }} className="ellipsis--noWrap d-inline-block w-100">
          show me the tooltip on hover
        </Button>
      </Tooltip>
    );
    const button = getByRole('button');
    fireEvent.mouseOver(button);
    const tooltipText = queryByText('A tooltip');
    expect(tooltipText).not.toBeInTheDocument();
  });

  it('should support custom class', () => {
    const { getByRole, getByTestId } = render(
      <Tooltip tooltip="A tooltip" className="custom-class">
        <Button>show me the tooltip on hover</Button>
      </Tooltip>
    );
    const button = getByRole('button');
    fireEvent.mouseOver(button);
    getByTestId('DesignSystem-Popover').classList.contains('custom-class');
  });
});

describe('Tooltip component with size prop', () => {
  it('should render tooltip with small size', () => {
    const { getByRole, getByTestId } = render(
      <Tooltip size="small" tooltip="A tooltip">
        <Button>show me the tooltip on hover</Button>
      </Tooltip>
    );
    const button = getByRole('button');
    fireEvent.mouseOver(button);
    const tooltipWrapper = getByTestId('DesignSystem-Tooltip-Wrapper');
    expect(tooltipWrapper).toHaveClass('Tooltip--small');
  });

  it('should render tooltip with regular size', () => {
    const { getByRole, getByTestId } = render(
      <Tooltip size="regular" tooltip="A tooltip">
        <Button>show me the tooltip on hover</Button>
      </Tooltip>
    );
    const button = getByRole('button');
    fireEvent.mouseOver(button);
    const tooltipWrapper = getByTestId('DesignSystem-Tooltip-Wrapper');
    expect(tooltipWrapper).not.toHaveClass('Tooltip--small');
    expect(tooltipWrapper).toHaveClass('Tooltip');
  });
});
