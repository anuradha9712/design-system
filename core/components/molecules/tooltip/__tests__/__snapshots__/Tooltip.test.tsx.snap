<<<<<<< HEAD
=======
// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip component 
  position: "bottom", size: "regular", tooltip: "Sample string"
 1`] = `
<body>
  <div>
    <div
      class="w-fit-content"
    >
      <div
        aria-expanded="false"
        aria-haspopup="dialog"
        class="w-fit-content"
        data-state="closed"
        triggerclass="w-fit-content"
      >
        <button
          class="Button Button--regular Button--basic Button--iconAlign-left"
          data-test="DesignSystem-Button"
          tabindex="0"
        >
          <span
            class="Button-text"
          >
            Button
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    data-floating-ui-portal=""
    id=":rn:"
  />
</body>
`;

exports[`Tooltip component 
  position: "bottom", size: "small", tooltip: "Sample string"
 1`] = `
<body>
  <div>
    <div
      class="w-fit-content"
    >
      <div
        aria-expanded="false"
        aria-haspopup="dialog"
        class="w-fit-content"
        data-state="closed"
        triggerclass="w-fit-content"
      >
        <button
          class="Button Button--regular Button--basic Button--iconAlign-left"
          data-test="DesignSystem-Button"
          tabindex="0"
        >
          <span
            class="Button-text"
          >
            Button
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    data-floating-ui-portal=""
    id=":rk:"
  />
</body>
`;

exports[`Tooltip component 
  position: "bottom-end", size: "regular", tooltip: "Sample string"
 1`] = `
<body>
  <div>
    <div
      class="w-fit-content"
    >
      <div
        aria-expanded="false"
        aria-haspopup="dialog"
        class="w-fit-content"
        data-state="closed"
        triggerclass="w-fit-content"
      >
        <button
          class="Button Button--regular Button--basic Button--iconAlign-left"
          data-test="DesignSystem-Button"
          tabindex="0"
        >
          <span
            class="Button-text"
          >
            Button
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    data-floating-ui-portal=""
    id=":r13:"
  />
</body>
`;

exports[`Tooltip component 
  position: "bottom-end", size: "small", tooltip: "Sample string"
 1`] = `
<body>
  <div>
    <div
      class="w-fit-content"
    >
      <div
        aria-expanded="false"
        aria-haspopup="dialog"
        class="w-fit-content"
        data-state="closed"
        triggerclass="w-fit-content"
      >
        <button
          class="Button Button--regular Button--basic Button--iconAlign-left"
          data-test="DesignSystem-Button"
          tabindex="0"
        >
          <span
            class="Button-text"
          >
            Button
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    data-floating-ui-portal=""
    id=":r10:"
  />
</body>
`;

exports[`Tooltip component 
  position: "bottom-start", size: "regular", tooltip: "Sample string"
 1`] = `
<body>
  <div>
    <div
      class="w-fit-content"
    >
      <div
        aria-expanded="false"
        aria-haspopup="dialog"
        class="w-fit-content"
        data-state="closed"
        triggerclass="w-fit-content"
      >
        <button
          class="Button Button--regular Button--basic Button--iconAlign-left"
          data-test="DesignSystem-Button"
          tabindex="0"
        >
          <span
            class="Button-text"
          >
            Button
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    data-floating-ui-portal=""
    id=":rt:"
  />
</body>
`;

exports[`Tooltip component 
  position: "bottom-start", size: "small", tooltip: "Sample string"
 1`] = `
<body>
  <div>
    <div
      class="w-fit-content"
    >
      <div
        aria-expanded="false"
        aria-haspopup="dialog"
        class="w-fit-content"
        data-state="closed"
        triggerclass="w-fit-content"
      >
        <button
          class="Button Button--regular Button--basic Button--iconAlign-left"
          data-test="DesignSystem-Button"
          tabindex="0"
        >
          <span
            class="Button-text"
          >
            Button
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    data-floating-ui-portal=""
    id=":rq:"
  />
</body>
`;

exports[`Tooltip component 
  position: "left", size: "regular", tooltip: "Sample string"
 1`] = `
<body>
  <div>
    <div
      class="w-fit-content"
    >
      <div
        aria-expanded="false"
        aria-haspopup="dialog"
        class="w-fit-content"
        data-state="closed"
        triggerclass="w-fit-content"
      >
        <button
          class="Button Button--regular Button--basic Button--iconAlign-left"
          data-test="DesignSystem-Button"
          tabindex="0"
        >
          <span
            class="Button-text"
          >
            Button
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    data-floating-ui-portal=""
    id=":r19:"
  />
</body>
`;

exports[`Tooltip component 
  position: "left", size: "small", tooltip: "Sample string"
 1`] = `
<body>
  <div>
    <div
      class="w-fit-content"
    >
      <div
        aria-expanded="false"
        aria-haspopup="dialog"
        class="w-fit-content"
        data-state="closed"
        triggerclass="w-fit-content"
      >
        <button
          class="Button Button--regular Button--basic Button--iconAlign-left"
          data-test="DesignSystem-Button"
          tabindex="0"
        >
          <span
            class="Button-text"
          >
            Button
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    data-floating-ui-portal=""
    id=":r16:"
  />
</body>
`;

exports[`Tooltip component 
  position: "right", size: "regular", tooltip: "Sample string"
 1`] = `
<body>
  <div>
    <div
      class="w-fit-content"
    >
      <div
        aria-expanded="false"
        aria-haspopup="dialog"
        class="w-fit-content"
        data-state="closed"
        triggerclass="w-fit-content"
      >
        <button
          class="Button Button--regular Button--basic Button--iconAlign-left"
          data-test="DesignSystem-Button"
          tabindex="0"
        >
          <span
            class="Button-text"
          >
            Button
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    data-floating-ui-portal=""
    id=":r1f:"
  />
</body>
`;

exports[`Tooltip component 
  position: "right", size: "small", tooltip: "Sample string"
 1`] = `
<body>
  <div>
    <div
      class="w-fit-content"
    >
      <div
        aria-expanded="false"
        aria-haspopup="dialog"
        class="w-fit-content"
        data-state="closed"
        triggerclass="w-fit-content"
      >
        <button
          class="Button Button--regular Button--basic Button--iconAlign-left"
          data-test="DesignSystem-Button"
          tabindex="0"
        >
          <span
            class="Button-text"
          >
            Button
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    data-floating-ui-portal=""
    id=":r1c:"
  />
</body>
`;

exports[`Tooltip component 
  position: "top", size: "regular", tooltip: "Sample string"
 1`] = `
<body>
  <div>
    <div
      class="w-fit-content"
    >
      <div
        aria-expanded="false"
        aria-haspopup="dialog"
        class="w-fit-content"
        data-state="closed"
        triggerclass="w-fit-content"
      >
        <button
          class="Button Button--regular Button--basic Button--iconAlign-left"
          data-test="DesignSystem-Button"
          tabindex="0"
        >
          <span
            class="Button-text"
          >
            Button
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    data-floating-ui-portal=""
    id=":r5:"
  />
</body>
`;

exports[`Tooltip component 
  position: "top", size: "small", tooltip: "Sample string"
 1`] = `
<body>
  <div>
    <div
      class="w-fit-content"
    >
      <div
        aria-expanded="false"
        aria-haspopup="dialog"
        class="w-fit-content"
        data-state="closed"
        triggerclass="w-fit-content"
      >
        <button
          class="Button Button--regular Button--basic Button--iconAlign-left"
          data-test="DesignSystem-Button"
          tabindex="0"
        >
          <span
            class="Button-text"
          >
            Button
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    data-floating-ui-portal=""
    id=":r2:"
  />
</body>
`;

exports[`Tooltip component 
  position: "top-end", size: "regular", tooltip: "Sample string"
 1`] = `
<body>
  <div>
    <div
      class="w-fit-content"
    >
      <div
        aria-expanded="false"
        aria-haspopup="dialog"
        class="w-fit-content"
        data-state="closed"
        triggerclass="w-fit-content"
      >
        <button
          class="Button Button--regular Button--basic Button--iconAlign-left"
          data-test="DesignSystem-Button"
          tabindex="0"
        >
          <span
            class="Button-text"
          >
            Button
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    data-floating-ui-portal=""
    id=":rh:"
  />
</body>
`;

exports[`Tooltip component 
  position: "top-end", size: "small", tooltip: "Sample string"
 1`] = `
<body>
  <div>
    <div
      class="w-fit-content"
    >
      <div
        aria-expanded="false"
        aria-haspopup="dialog"
        class="w-fit-content"
        data-state="closed"
        triggerclass="w-fit-content"
      >
        <button
          class="Button Button--regular Button--basic Button--iconAlign-left"
          data-test="DesignSystem-Button"
          tabindex="0"
        >
          <span
            class="Button-text"
          >
            Button
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    data-floating-ui-portal=""
    id=":re:"
  />
</body>
`;

exports[`Tooltip component 
  position: "top-start", size: "regular", tooltip: "Sample string"
 1`] = `
<body>
  <div>
    <div
      class="w-fit-content"
    >
      <div
        aria-expanded="false"
        aria-haspopup="dialog"
        class="w-fit-content"
        data-state="closed"
        triggerclass="w-fit-content"
      >
        <button
          class="Button Button--regular Button--basic Button--iconAlign-left"
          data-test="DesignSystem-Button"
          tabindex="0"
        >
          <span
            class="Button-text"
          >
            Button
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    data-floating-ui-portal=""
    id=":rb:"
  />
</body>
`;

exports[`Tooltip component 
  position: "top-start", size: "small", tooltip: "Sample string"
 1`] = `
<body>
  <div>
    <div
      class="w-fit-content"
    >
      <div
        aria-expanded="false"
        aria-haspopup="dialog"
        class="w-fit-content"
        data-state="closed"
        triggerclass="w-fit-content"
      >
        <button
          class="Button Button--regular Button--basic Button--iconAlign-left"
          data-test="DesignSystem-Button"
          tabindex="0"
        >
          <span
            class="Button-text"
          >
            Button
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    data-floating-ui-portal=""
    id=":r8:"
  />
</body>
`;

exports[`Tooltip component should not render tooltip when showTooltip is false 1`] = `
<body>
  <div>
    <button
      class="Button Button--regular Button--basic Button--iconAlign-left"
      data-test="DesignSystem-Button"
      tabindex="0"
    >
      <span
        class="Button-text"
      >
        Button
      </span>
    </button>
  </div>
</body>
`;
>>>>>>> e56ca8cd (feat(popover): add new popover component)
